src/utilities/observables/asyncMap.ts

```diff
@@ -33,20 +33,21 @@ export function asyncMap<V, R>(
             .then(both, both)
             .then(
               (result) => {
-                --activeCallbackCount;
                 next && next.call(observer, result);
-                if (completed) {
-                  handler.complete!();
-                }
               },
               (error) => {
-                --activeCallbackCount;
                 throw error;
               }
             )
             .catch((caught) => {
               error && error.call(observer, caught);
             });
+          promiseQueue.finally(() => {
+            --activeCallbackCount;
+            if (completed) {
+              handler.complete!();
```

Comment:
```suggestion
          promiseQueue.finally(() => {
            --activeCallbackCount;
            // Because of the .finally delay, it's possible handler.complete() gets
            // called shortly before activeCallbackCount falls to 0, so each time we
            // decrement activeCallbackCount we give handler.complete() another
            // chance to call observer.complete().
            if (completed) {
              handler.complete!();
```
This might be a helpful comment, given the racy nature of this code.